### ISSUE TITLE
Display select_account prompt when noaccountyet

### ISSUE THREAD

**Issue by tonyskalski** [mouneyrac/moodle-auth_googleoauth2#208]
We have Moodle auto-provisioning of accounts turned off. In a clean browser session, if a user logs in with a non-institution Google account and then attempts to login to Moodle via auth_googleoauth2, they receive an error. We have temporarily worked around this by including a link to accounts.google.com/addsession in the error message.

A better fix would be to handle this noaccountyet exception and re-attempt Google oauth with the prompt=select_account option set, to force the display of the account picker.

Thanks!

**Comment by tonyskalski** [comment]
This would be something like the OIDCAuthRequestParams option in the openid connect apache module. See https://github.com/pingidentity/mod_auth_openidc/blob/master/auth_openidc.conf.

By being able to pass additional parameters on the authn request URL, we could set prompt=select_account.

**Comment by mouneyrac** [comment]
Hi Tony, it seems to be an interesting issue but I am not 100% sure to understand. Would you have screenshot/video and detailled information about the error message?

Basically from what I understand you are saying that a user has two Google accounts (one personal and one from your school), the user doesn't pick the school one and then he can not access Moodle (because automatic account creation has been disabled in Moodle)? 

Alternatively: he is already logged in Google with the wrong account and so he can not login in Moodle ?

That something I never tested (both use cases are actually the same). 

Can you confirm that it is the problem?

**Comment by tonyskalski** [comment]
Yes - the use cases are the same. I don't think there is anything we can do about the user choosing the wrong account. We are more concerned with a user already logged in to a non-school account.

I have since learned that you are using [https://github.com/thephpleague/oauth2-client](url) and have modified that part of the code to suit our needs:

[]# diff AbstractProvider.php{,.orig}
132a133
>             'approval_prompt' => isset($options['approval_prompt']) ? $options['approval_prompt'] : 'auto',
[root@moodle-2016-17-test Provider]# diff Google.php{,.orig}
50,65d49
<     /**
<      * @var string If set, this will be sent to google as the "prompt" parameter.
<      * @link https://developers.google.com/identity/protocols/OpenIDConnect#prompt
<      */
<     public $promptType = 'select_account';
< 
<     public function setPromptType($prompt_type)
<     {
<     	$this->promptType = $prompt_type;
<     }
< 
<     public function getPromptType()
<     {
<     	return $this->promptType;
<     }
< 
137,140d120
< 
< 	if (!empty($this->promptType)) {
< 		$url .= '&' . $this->httpBuildQuery(['prompt'=> $this->promptType]);
< 	}

Of course, this means it is not your code.

**Comment by mouneyrac** [comment]
Ah thanks for sharing the fix, I'll add it to next version (hopefully Ill find some time to work on the plugin - I suppose beginning of next year - I estimate to about 2 weeks of work on it to reach version 3.0)
